library/nxos_get_interface.py

```diff
@@ -0,0 +1,689 @@
+        vrf_table = None
+
+    key_map = {
+                "prefix": "ip_addr",
+                "masklen": "mask",
+                "subnet": "subnet",
+            }
+
+    interface_info = apply_key_map(key_map, interface_table)
+    interface_info['type'] = interface_type
+    interface_info['interface'] = interface
+
+    if vrf_table is not None:
+        vrf = vrf_table['vrf-name-out']
+        interface_info['vrf'] = vrf
+
+    return interface_info
+
+
+def get_layer2_info(device, port, module):
+    """Gets current config of L2 switchport
+    Args:
+        device (Device): This is the device object of an NX-API enabled device
+            using the Device class within device.py
+        port (str): full name of interface, i.e. Ethernet1/1
+    Returns:
+        dictionary with k/v pairs for L2 vlan config
+    """
+
+    command = 'show interface {0} switchport'.format(port)
+
+    body = parsed_data_from_device(device, command, module)
+
+    if body:
+        key_map = {
+            "interface": "interface",
+            "oper_mode": "mode",
+            "switchport": "switchport",
+            "access_vlan": "access_vlan",
+            "access_vlan_name": "access_vlan_name",
+            "native_vlan": "native_vlan",
+            "native_vlan_name": "native_vlan_name",
+            "trunk_vlans": "trunk_vlans"
+        }
+
+        layer2_table = body['TABLE_interface']['ROW_interface']
+
+        layer2_info = apply_key_map(key_map, layer2_table)
+
+        return layer2_info
+    else:
+        return {}
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            interface=dict(required=True),
+            detail=dict(choices=BOOLEANS, type='bool'),
+            protocol=dict(choices=['http', 'https'], default='http'),
+            host=dict(required=True),
+            username=dict(type='str'),
+            password=dict(type='str'),
+        ),
+        supports_check_mode=False
+    )
+    if not HAS_PYCSCO:
+        module.fail_json(msg='There was a problem loading pycsco')
+
+    auth = Auth(vendor='cisco', model='nexus')
+    username = module.params['username'] or auth.username
+    password = module.params['password'] or auth.password
+    protocol = module.params['protocol']
+    detail = module.params['detail'] or False
+    host = socket.gethostbyname(module.params['host'])
+    interface = module.params['interface'].lower()
+
+    device = Device(ip=host, username=username, password=password,
+                    protocol=protocol)
+
+    interface_type = get_interface_type(interface)
+    normalized_interface = normalize_interface(interface)
+
+    if interface_type == 'unknown':
+        module.fail_json(
+            msg='unknown interface type found-1',
+            interface=interface)
+
+    resource, mode, interface_table = get_resource(
+                    device, normalized_interface, interface_type, module)
+
+    layer3_info = None
+    layer2_info = None
+    oper = None
+
```

Comment:
Instead of assigning these the value of `None`, use `{}` if they are going to be dictionaries...`[]` for lists, etc.